sdk/android/ooyalaSkinSDK/src/androidTest/java/com/ooyala/android/ooyalaskinsdk/JSONDeepMergeTest.java

```diff
@@ -0,0 +1,176 @@
+      assertFalse("JSON Exception occurred", true);
+    }
+
+    assertNull(initial);
+    assertNotNull(overrides);
+  }
+
+  public void testNullOverrides() {
+    JSONObject initial = new JSONObject();
+    JSONObject overrides = null;
+    try {
+      initial.put("first", "first");
+      initial.put("second", 2);
+      JSONDeepMerge.inPlaceDeepMerge(initial, overrides);
+    } catch (JSONException e) {
+      assertFalse("JSON Exception occurred", true);
+    }
+
+    try {
+      assertEquals(2, initial.length());
+      assertEquals("first", initial.getString("first"));
+      assertEquals(2, initial.getInt("second"));
+      assertNull(overrides);
+    } catch (JSONException e) {
+      assertFalse("JSON Exception occurred", true);
+    }
+  }
+  public void testNoOverlap() {
+    JSONObject initial = new JSONObject();
+    JSONObject overrides = new JSONObject();
+    try {
+      initial.put("first", "first");
+      overrides.put("second", "second");
+      overrides.put("third", 3);
+      JSONDeepMerge.inPlaceDeepMerge(initial, overrides);
+    } catch (JSONException e) {
+      assertFalse("JSON Exception occurred", true);
+    }
+
+    try {
+      assertEquals(3, initial.length());
+      assertEquals("first", initial.getString("first"));
+      assertEquals("second", initial.getString("second"));
+      assertEquals(3, initial.getInt("third"));
+    } catch (JSONException e) {
+      assertFalse("JSON Exception occurred", true);
+    }
+  }
+
+  public void testWithOverlap() {
+    JSONObject initial = new JSONObject();
+    JSONObject overrides = new JSONObject();
+    try {
+      initial.put("first", "first");
+      overrides.put("first", "overridden");
+      overrides.put("second", "second");
+      JSONDeepMerge.inPlaceDeepMerge(initial, overrides);
+    } catch (JSONException e) {
+      assertFalse("JSON Exception occurred", true);
+    }
+
+    try {
+      assertEquals(2, initial.length());
+      assertNotSame("first", initial.getString("first"));
+      assertEquals("overridden", initial.getString("first"));
+    } catch (JSONException e) {
+      assertFalse("JSON Exception occurred", true);
+    }
+  }
+
+  public void testWithOverlapDifferentType() {
+    JSONObject initial = new JSONObject();
+    JSONObject overrides = new JSONObject();
+    JSONObject child = new JSONObject();
+    try {
+      initial.put("first", "first");
+      overrides.put("first", 1);
+
+      initial.put("second", "second");
+      child.put("inchild", "inchild");
+      overrides.put("second", child);
+
+      JSONDeepMerge.inPlaceDeepMerge(initial, overrides);
+    } catch (JSONException e) {
+      Log.e("JSONDeepMergeTest", "JSON Exception", e);
+      assertFalse("JSON Exception occurred", true);
+    }
+
+    assertEquals(2, initial.length());
+    try {
+      assertNotSame("first", initial.getString("first"));
+    } catch (JSONException e) {
+      // Supposed to fail, first is no longer a string;
```

Comment:
it's supposed to fail, but it doesn't fail? I think the assert is correct, as "first" (String) is not the same as the Integer object in the initial "first" key.